Source/JavaScriptCore/heap/MarkedBlock.cpp

```diff
@@ -210,17 +210,57 @@ void MarkedBlock::Handle::resumeAllocating(FreeList& freeList)
     sweep(&freeList);
 }
 
+#if ENABLE(MARKEDBLOCK_TEST_DUMP_INFO)
```

Comment:
typo in the commit msg: /invokation/invocation/ and /interpretted/interpreted/.

Please also add a comment explaining why `m_lock` "will always be non-zero" so that it's understood the code path taken to reach the diagnostics will acquire the lock thereby setting it.  It's been set is not an indication that the block hasn't been inadvertently cleared by some external agent.

Reply:
> Please also add a comment explaining why m_lock "will always be non-zero"

There is a code comment already explaining m_lock, but I'll enhance the commit message, assuming that's what you mean.

> It's been set is not an indication that the block hasn't been inadvertently cleared by some external agent.

Of course. I'm not sure which part of the commit message implies this.

Oh, maybe this, which I can clarify:

> locations on the page that are written before corruption is detected.

I'll say instead: "... may have been written after the corruption occurs."